dexs/plunderswap/index.ts

```diff
@@ -0,0 +1,49 @@
+import fetchURL from "../../utils/fetchURL"
+import { ChainBlocks, FetchOptions, SimpleAdapter } from "../../adapters/types";
+import { getUniqStartOfTodayTimestamp } from "../../helpers/getUniSubgraphVolume";
+
+const historicalVolumeEndpoint = "https://static.plunderswap.com/volume-history"
+
+interface IVolumeall {
+  value: string;
+  time: string;
+}
+
+const fetch = async (timestamp: number, _: ChainBlocks, { createBalances, startOfDay, }: FetchOptions) => {
+  const dailyVolume = createBalances()
+  const dayTimestamp = getUniqStartOfTodayTimestamp(new Date(timestamp * 1000))
+  const historicalVolume: IVolumeall[] = (await fetchURL(historicalVolumeEndpoint));
+  
+  // Find the entry closest to the start of day
+  const closestEntry = historicalVolume
+    .reduce((closest, current) => {

```

Comment:
i think you can filter time more than start time and less than end time instead before reduce